### PR TITLE
Connecticut Subtraction for contributions to qualified state tuition program

### LINKED ISSUE #2925: Connecticut Subtraction for contributions to qualified state tuition program
New additions for CT tax modeling

### PR DESCRIPTION
**Pull request by vrathi101**
Fixes #2925

### Summary
📱🌎💰

Add a new policy parameter `ubi_amount` to the US tax and benefit system. Implement a basic universal basic income (UBI) reform that replaces the standard deduction and personal exemption with a flat monthly payment. Update the `reform_us.yaml` file and the `policyengine-us.ipynb` notebook to demonstrate the UBI reform.

> _`is_valid` checks_
> _refactored for clarity_
> _autumn leaves falling_

### Walkthrough
*  Add a new parameter `child_allowance.amount` to the `parameters.yaml` file, which sets the monthly amount of the child allowance benefit per child ()

### CODE REVIEW

**Comment by vrathi101** [comment]
Just wanted to check if what I have so far is good before adding test files.

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+- bump: minor
+  changes:
+    added:
+    - Connecticut state tuition subtractions
+  date: 2023-09-05 18:48:29
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:5]:
```suggestion
```

File: changelog_entry.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+- bump: minor
+  changes:
+    added:
+    - Connecticut state tuition subtractions
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Connecticut state tuition subtraction.
```

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/tuition/max_amount.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+description: Connecticut provides a state tuitition subtraction under certain qualifications and for a maximum limit.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/tuition/max_amount.yaml:1]:
```suggestion
description: Connecticut limits the state tuition subtraction to this amount, based on filing status.
```

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/tuition/max_amount.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+metadata:
+  period: year
+  unit: currency-USD
+  label: Connecticut state tuition subtractions
+  breakdown:
+    - filing_status
+  reference:
+    - title: Connecticut General Statutes, Chapter 229, Sec. 12-701a
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/tuition/max_amount.yaml:9]: add tax form reference for 2022

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/tuition/max_amount.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+description: Connecticut provides a state tuitition subtraction under certain qualifications and for a maximum limit.
+metadata:
+  period: year
+  unit: currency-USD
+  label: Connecticut state tuition subtractions
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/tuition/max_amount.yaml:5]:
```suggestion
  label: Connecticut state tuition subtraction max amount
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/tuition/ct_tuition_subtractions.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    definition_period = YEAR
+    reference = "https://www.cga.ct.gov/current/pub/chap_229.htm#sec_12-701a"
+    defined_for = StateCode.CT
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.ct.tax.income.subtractions.tuition.max_amount
+        filing_status = tax_unit("filing_status", period)
+        max_amount = p[filing_status]
+        tuition = add(tax_unit, period, ["qualified_tuition_expenses"])
+        subtraction = min(tuition, max_amount)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/tuition/ct_tuition_subtractions.py:18]:
```suggestion
        return min_(tuition, max_amount)
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/tuition/ct_tuition_subtractions.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    value_type = float
+    entity = TaxUnit
+    label = "CT tuition subtractions"
+    unit = USD
+    definition_period = YEAR
+    reference = "https://www.cga.ct.gov/current/pub/chap_229.htm#sec_12-701a"
+    defined_for = StateCode.CT
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.ct.tax.income.subtractions.tuition.max_amount
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/tuition/ct_tuition_subtractions.py:14]:
```suggestion
        p = parameters(period).gov.states.ct.tax.income.subtractions.tuition
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/tuition/ct_tuition_subtractions.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+        filing_status = tax_unit("filing_status", period)
+        max_amount = p[filing_status]
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/tuition/ct_tuition_subtractions.py:16]:
```suggestion
        max_amount = p.max_amount[filing_status]
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/tuition/ct_tuition_subtractions.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
+from policyengine_us.model_api import *
+
+
+class ct_tuition_subtractions(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "CT tuition subtractions"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/tuition/ct_tuition_subtractions.py:7]:
```suggestion
    label = "Connecticut tuition subtraction"
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/tuition/ct_tuition_subtractions.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/tuition/ct_tuition_subtractions.py:1]: need unit test

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/tuition/ct_tuition_subtractions.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
+from policyengine_us.model_api import *
+
+
+class ct_tuition_subtractions(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/tuition/ct_tuition_subtractions.py:4]:
```suggestion
class ct_tuition_subtraction(Variable):
```

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/tuition/max_amount.yaml [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+  period: year
+  unit: currency-USD
+  label: Connecticut state tuition subtraction max amount
+  breakdown:
+    - filing_status
+  reference:
+    - title: Connecticut General Statutes, Chapter 229, Sec. 12-701a
+      href: https://www.cga.ct.gov/current/pub/chap_229.htm#sec_12-701a
+    - title: 2022 Form CT-1040 Connecticut Resident Income Tax Return Instructions
+      href: https://portal.ct.gov/-/media/DRS/Forms/2022/Income/2022-CT-1040-Instructions_1222.pdf#page21
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/tuition/max_amount.yaml:12]:
```suggestion
      href: https://portal.ct.gov/-/media/DRS/Forms/2022/Income/2022-CT-1040-Instructions_1222.pdf#page=21
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/tuition/ct_tuition_subtraction.py [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+    unit = USD
+    definition_period = YEAR
+    reference = "https://www.cga.ct.gov/current/pub/chap_229.htm#sec_12-701a"
+    defined_for = StateCode.CT
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.ct.tax.income.subtractions.tuition
+        filing_status = tax_unit("filing_status", period)
+        max_amount = p.max_amount[filing_status]
+        tuition = add(tax_unit, period, ["qualified_tuition_expenses"])
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/tuition/ct_tuition_subtraction.py:17]:
```suggestion
        # Qualified tuition expenses as defined under Section 529(b) of the Internal Revenue Code
        tuition = add(tax_unit, period, ["qualified_tuition_expenses"])
```

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need a git pull upstream master

Comment (PavelMakarchuk) [reply changelog_entry.yaml:1]: after that can request maxes review - resolve this comment

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/tuition/max_amount.yaml [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+  unit: currency-USD
+  label: Connecticut state tuition subtraction max amount
+  breakdown:
+    - filing_status
+  reference:
+    - title: Connecticut General Statutes, Chapter 229, Sec. 12-701a
+      href: https://www.cga.ct.gov/current/pub/chap_229.htm#sec_12-701a
+    - title: 2022 Form CT-1040 Connecticut Resident Income Tax Return Instructions
+      href: https://portal.ct.gov/-/media/DRS/Forms/2022/Income/2022-CT-1040-Instructions_1222.pdf#page=21
+SINGLE:
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/tuition/max_amount.yaml:13]: need head of household

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/tuition/max_amount.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/tuition/max_amount.yaml:1]: rename `cap.yaml`

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**